Arm64 BFloat included when no SVE or BFloat ISA available

When building with BFloat16 extensions enabled, the NEOVERSEN2 target is including the sbgemm kernels even when there is no SVE/BFloat ISA support.

This can be seen in the Julia build logs for our attempt to build: https://buildkite.com/julialang/yggdrasil/builds/7890#018d31ff-56ad-42c4-b883-5b705218ca05/6-2849

In this case, the errors we see are:
```
[17:25:10] In file included from ../kernel/arm64/sbgemm_kernel_8x4_neoversen2.c:29:
[17:25:10] /opt/x86_64-linux-musl/lib/clang/13.0.1/include/arm_sve.h:15:2: error: "SVE support not enabled"
[17:25:10] #error "SVE support not enabled"
[17:25:10]  ^
```
and
```
[17:25:10] ../kernel/arm64/sbgemm_ncopy_4_neoversen2.c:40:3: error: use of undeclared identifier 'svbool_t'
[17:25:10] In file included from   svbool_t pg16 = svdupq_b16(1, 1, 1, 1, 0, 0, 0, 0);
[17:25:10]   ^
[17:25:10] ../kernel/arm64/sbgemm_kernel_8x4_neoversen2.c:34:
[17:25:10] ../kernel/arm64/sbgemm_kernel_8x4_neoversen2_impl.c:73:3: error: unknown type name 'svbfloat16_t'; did you mean 'bfloat16'?
[17:25:10]   svbfloat16_t ma0, ma1, ma2, ma3, mb0, mb1;
[17:25:10]   ^~~~~~~~~~~~
[17:25:10]   bfloat16
```

The command line that is being used to compile is
```
[17:25:10] cc -O2 -DMAX_STACK_ALLOC=2048 -Wall -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=512 -DMAX_PARALLEL_NUMBER=1 -DBUILD_BFLOAT16 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.26\" -march=armv8.2-a -mtune=cortex-a72 -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=_sbgemm_kernel_NEOVERSEN2 -DASMFNAME=_sbgemm_kernel_NEOVERSEN2_ -DNAME=sbgemm_kernel_NEOVERSEN2_ -DCNAME=sbgemm_kernel_NEOVERSEN2 -DCHAR_NAME=\"sbgemm_kernel_NEOVERSEN2_\" -DCHAR_CNAME=\"sbgemm_kernel_NEOVERSEN2\" -DNO_AFFINITY -DTS=_NEOVERSEN2 -I.. -DBUILD_KERNEL -DTABLE_NAME=gotoblas_NEOVERSEN2 -DBFLOAT16 -UDOUBLE  -UCOMPLEX -c -DBFLOAT16 -UDOUBLE -UCOMPLEX ../kernel/arm64/sbgemm_kernel_8x4_neoversen2.c -o sbgemm_kernel_NEOVERSEN2.o
```
So it is getting the options `-march=armv8.2-a -mtune=cortex-a72`, however the SVE and BFloat16 extensions are not part of the armv8.2-a architecture.

@giordano did some digging and points to the conditional here https://github.com/OpenMathLib/OpenBLAS/blob/d6a5174e9c50b9f68db96d7d7818f92cdfb4e7ec/Makefile.arm64#L138-L142 as being the likely problem, and I think I agree with that assessment.

I think what probably needs to happen is the NeoverseN2 target needs to fallback to the generic BFloat16 kernels when it hits that case instead of using the optimized ones.

--- a/Makefile.arm64
+++ b/Makefile.arm64
@@ -116,13 +116,13 @@ endif
 endif
 endif
 else
-CCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a72
+CCOMMON_OPT += -march=armv8.2-a+sve -mtune=cortex-a72
 ifneq ($(F_COMPILER), NAG)
 FCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a72
 endif
 endif
 else
-CCOMMON_OPT += -march=armv8-a -mtune=cortex-a72
+CCOMMON_OPT += -march=armv8-a+sve -mtune=cortex-a72
 ifneq ($(F_COMPILER), NAG)
 FCOMMON_OPT += -march=armv8-a -mtune=cortex-a72
 endif
@@ -138,7 +138,7 @@ ifeq (1, $(filter 1,$(GCCMINORVERSIONGTEQ4) $(GCCVERSIONGTEQ11) $(ISCLANG)))
 ifneq ($(OSNAME), Darwin)
 CCOMMON_OPT += -march=armv8.5-a+sve+sve2+bf16 -mtune=neoverse-n2
 else
-CCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a72
+CCOMMON_OPT += -march=armv8.2-a+sve -mtune=cortex-a72
 endif
 ifneq ($(F_COMPILER), NAG)
 FCOMMON_OPT += -march=armv8.5-a+sve+sve2+bf16 -mtune=neoverse-n2
@@ -156,13 +156,13 @@ endif
 endif
 endif
 else
-CCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a72
+CCOMMON_OPT += -march=armv8.2-a+sve -mtune=cortex-a72
 ifneq ($(F_COMPILER), NAG)
 FCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a72
 endif
 endif
 else
-CCOMMON_OPT += -march=armv8-a -mtune=cortex-a72
+CCOMMON_OPT += -march=armv8-a+sve -mtune=cortex-a72
 ifneq ($(F_COMPILER), NAG)
 FCOMMON_OPT += -march=armv8-a -mtune=cortex-a72
 endif

